object/user_util.go

```diff
@@ -268,6 +268,9 @@ func CheckPermissionForUpdateUser(oldUser, newUser *User, isAdmin bool, lang str
 	}
 	if oldUser.DisplayName != newUser.DisplayName {
 		item := GetAccountItemByName("Display name", organization)
+		if pass, err := CheckAccountItemRegex(item, newUser.DisplayName, lang); !pass {
```

Comment:
Refer to:

![image](https://github.com/casdoor/casdoor/assets/3787410/7cee5128-f85f-4175-9f4b-1c31b616bb3d)